### PR TITLE
collect provision_duration metrics

### PR DESCRIPTION
**Pull request by life0215**
**Description**

Changes proposed in this pull request:

- add ProvisioningSucceeded event
- For provision operation duration, instead of subscribing to every provision step,
it will subscribe to provision ProvisioningSucceeded event(sent from last step of last provision stage which finished without issue)

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/internal/process/provisioning/staged_manager.go [hunk 1]
```diff
@@ -133,6 +133,15 @@ func (m *StagedManager) Execute(operationID string) (time.Duration, error) {
...
 			}
+
+			// if last step for the last stage finished wihtout issue, set state to domain.Succeeded and publish event
+			if i+1 == len(m.stages) && j+1 == len(stage.steps) {
+				opeartion := processedOperation
+				opeartion.State = domain.Succeeded
+				m.publisher.Publish(context.TODO(), process.ProvisioningSucceeded{
+					Operation: opeartion,
+				})
+			}
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/provisioning/staged_manager.go:144]: you can send an event outside of the loop, there is a piece of code, which sets the state Succeeded: https://github.com/kyma-project/control-plane/blob/main/components/kyma-environment-broker/internal/process/provisioning/staged_manager.go#L144
so you don't need to check if it is the last step.

Comment (life0215) [reply components/kyma-environment-broker/internal/process/provisioning/staged_manager.go]: Thanks  @piotrmiskiewicz  for the advice, has done the change. Please kindly review.

**Review by piotrmiskiewicz — APPROVED** [review]

File: components/kyma-environment-broker/internal/process/provisioning/staged_manager_test.go [hunk 2]
```diff
@@ -189,3 +190,70 @@ func (h *CollectingEventHandler) AssertProcessedSteps(t *testing.T, stepNames []
...
+	// given
+	const opID = "op-0001234"
+	operation := FixProvisionOperation("op-0001234")
+	mgr, _, pubSub := SetupStagedManager2(operation)
+	mgr.AddStep("stage-1", &testingStep{name: "first", eventPublisher: pubSub})
+
+	rc := &resultCollector{}
+	rc.duration = 123
+	pubSub.Subscribe(process.ProvisioningSucceeded{}, rc.OnProvisionSucceed)
+	fmt.Printf("rc: %.4f \n", rc.duration)
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/provisioning/staged_manager_test.go:251]: remove this printf

File: components/kyma-environment-broker/internal/process/provisioning/staged_manager_test.go [hunk 3]
```diff
@@ -189,3 +190,70 @@ func (h *CollectingEventHandler) AssertProcessedSteps(t *testing.T, stepNames []
...
+	assert.Equal(t, domain.Succeeded, rc.state)
+	return nil
+}
+
+func (rc *resultCollector) AssertDurationGreaterThanZero(t *testing.T) error {
+	assert.Greater(t, rc.duration, 0.0)
+	return nil
+}
+
+func SetupStagedManager2(op internal.ProvisioningOperation) (*provisioning.StagedManager, storage.Operations, *event.PubSub) {
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/provisioning/staged_manager_test.go:227]: find a better name - not just suffix "2"

**Pull request merged**